backend/director/agents/scene_index.py

```
@@ -0,0 +1,120 @@
+    def __init__(self, session: Session, **kwargs):
+        self.agent_name = "scene_index"
+        self.description = (
+            "Retrieve and display the indexed scene descriptions of a video. "
+            "Use this when the user wants to view, see, or list the scene descriptions "
+            "or scene index of a video that has already been scene-indexed."
+        )
+        self.parameters = SCENE_INDEX_AGENT_PARAMETERS
+        super().__init__(session=session, **kwargs)
+
+    def run(
+        self, video_id: str, collection_id: str, scene_index_id: str = None
```

Ruff (0.15.10)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)